### PR TITLE
Version mypy=1.4.1 falsly reports unused ignore comment

### PR DESCRIPTION
**Pull request by Felixoid**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Attempt to fix style-check by rollback mypy to 1.4.0

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
This is an automated comment for commit a7968d5d6d15e36f4ed21a25c5e7d52c426e7b44 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/51769/a7968d5d6d15e36f4ed21a25c5e7d52c426e7b44/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: docker/test/style/Dockerfile [hunk 1]
```diff
@@ -18,7 +18,7 @@ RUN apt-get update && env DEBIAN_FRONTEND=noninteractive apt-get install --yes \
     python3-pip \
     shellcheck \
     yamllint \
-    && pip3 install black==23.1.0 boto3 codespell==2.2.1 dohq-artifactory mypy PyGithub unidiff pylint==2.6.2 \
+    && pip3 install black==23.1.0 boto3 codespell==2.2.1 dohq-artifactory mypy==1.4.0 PyGithub unidiff pylint==2.6.2 \
```
Comment (alexey-milovidov) [inline docker/test/style/Dockerfile:21]: Also, we have to pin boto3, dohq-artifactory, PyGithub, unidiff.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by qoega** [comment]
Still
```
Errors while processing ../..//tests/ci/:
../../tests/ci/github_helper.py:114: error: Unused "type: ignore" comment
../../tests/ci/commit_status_helper.py:10: error: Unused "type: ignore" comment
../../tests/ci/merge_pr.py:157: error: Unused "type: ignore" comment
Found 3 errors in 3 files (checked 54 source files)
```

I think those are valid errors.

**Comment by qoega** [comment]
>PullRequestCI / StyleCheck (pull_request) Successful in 4m

I understand that we want green PR first and only then merge. But maybe you want to have working style check in other PR and we need to merge it ASAP?

**Pull request merged**